lib/dialects/db2/index.js

```diff
@@ -62,4 +63,7 @@ Db2Dialect.prototype.TICK_CHAR_LEFT = '"';
 Db2Dialect.prototype.TICK_CHAR_RIGHT = '"';
 Db2Dialect.prototype.DataTypes = DataTypes;
 
+// register dialect implementation
+registerDialectImpl('db2', Db2Dialect);
```

Comment:
I am not a fan of imports with side effects.

I think it would be better if users did

```typescript
import Sequelize from 'sequelize';
import MyCustomDialect from 'my-custom-dialect-package';

Sequelize.registerDialect('dialectName', MyCustomDialect);
```

instead of

```typescript
import Sequelize from 'sequelize';
import 'my-custom-dialect-package';
```

It's slightly more verbose but it's explicit that something is happening.

The `sequelize.js` file can still register the existing dialect for backward compatibility of course.